### PR TITLE
JAMES-2958 Limit domain name size to not longer than 255 characters

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[ERROR]   SmtpRandomStoringTest.oneHundredMailsShouldHaveBeenStoredBetweenFourAndEightTimes:168 » ConditionTimeout
```
test this please

**Comment by Arsnael** [comment]
> In fact it should be 253.

? https://tools.ietf.org/html/rfc5321#section-4.5.3.1.2

**Comment by Arsnael** [comment]
> Sorry, I found this in an older RFC.

No problem, it's always a mess with all those RFC updates :)

**Comment by chibenwa** [comment]
Merged

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java [hunk 1]
```diff
@@ -108,8 +107,7 @@ void addDomainMappingShouldRespondWithNoContent() {
             .when()
                 .put("from.com")
             .then()
-                .statusCode(HttpStatus.NO_CONTENT_204)
-                .body(isEmptyString());
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java]: nothing prevent an HTTP server to say that there's no body and return some. This check makes sense

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java [hunk 2]
```diff
@@ -95,30 +94,24 @@ void setUp() throws Exception {
...
-                    .jsonPath()
-                    .getList(".");
-
-            assertThat(domains).isEmpty();
+            given()
+                .get()
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .body(".", hasSize(0));
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java]: We should discuss this: I personally prefer extract + assertj has we don't have to learn a second assertion library and obviously matcher objects are way inferiors to builder pattern (it's the reason why we stopped using junit assertion for assertj in the first place).
Furthermore, I profiled our test execution and rest-assured is a slow library. I once started to remove it from our code base. I don't say we should do that now but relying more on it doesn't seem the right thing to do.

Comment (Arsnael) [reply server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java]: I never really knew much the difference between the two to be honest. I just thought this way is less verbose. But if you tell me you checked and this way is slower than extract and assertj then I will pay more attention to follow this pattern next time. Thanks for the feedback

**Pull request closed without merging**